pkg/printer/printer.go

```diff
@@ -191,25 +192,39 @@ func GetFlowType(f v1.Flow) string {
 
 // WriteProtoFlow writes v1.Flow into the output writer.
 func (p *Printer) WriteProtoFlow(res *observerpb.GetFlowsResponse) error {
+	var err error
+	write := func(w io.Writer, a ...interface{}) {
+		if err != nil {
+			return
+		}
+		_, err = fmt.Fprint(w, a...)
+	}
```

Comment:
For what it's worth, I'm fine with the code as is. Yes, the printer package could use a refactor to make error handling more pleasant, but that's out of scope for this PR. Let's not have a discussion about our error handling paradigms in Go in a PR which only attempts to add a simple feature.

Reply:
> Checking for err before setting it is odd. After a careful read, I think I understand the idea: once err is not nil then write becomes a no-op, and we only have to check and report the error one we're done invoking write. This is bit too "cute" for my taste, and I would expect at least a comment.

FWIW, this is not my own invention but a pattern described by Rob Pike in this [blog post from 2015](https://blog.golang.org/errors-are-values) to avoid repetitive error-checking and I think this applies quite nicely here as it allows removing a whole bunch of error-checking while keeping the code readable.

Reply:
Thanks for the link!